### PR TITLE
 [h5cpp] Add new port

### LINKED ISSUE #11744: [New Port Request] h5cpp
Library name: h5cpp

Library description: library to interact with h5 files

Source repository URL: https://github.com/steven-varga/h5cpp

Project homepage (if different from the source repository): http://h5cpp.org/

Anything else that is useful to know when adding (such as optional features the library may have that should be included):

### PR DESCRIPTION
**Pull request by nilox94**
**Describe the pull request**

- What does your PR fix? Fixes #11744

- Which triplets are supported/not supported? Have you updated the CI baseline?
Not yet

- Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?
Yes

### CODE REVIEW

File: ports/h5cpp/portfile.cmake [hunk 1]
```diff
@@ -0,0 +1,27 @@
+set(VCPKG_POLICY_EMPTY_PACKAGE enabled)
```
Comment (JackBoosY) [inline ports/h5cpp/portfile.cmake:1]: Why add this line?

File: ports/h5cpp/usage [hunk 2]
```diff
@@ -0,0 +1,4 @@
+The package h5cpp provides CMake targets:
```
Comment (JackBoosY) [inline ports/h5cpp/usage:1]: The usage should be auto generated, so we don't need this.

**Review by JackBoosY — CHANGES_REQUESTED** [review]

File: ports/h5cpp/portfile.cmake (hunk 1 above)
Comment (nilox94) [reply ports/h5cpp/portfile.cmake:1]: No reason at all.
I based this portfile on another one containing this line.
I thought it had something to do with header-only libraries but now I see this library doesn't need it.

**Comment by JackBoosY** [comment]
Please use my changes: 
[h5cpp.zip](https://github.com/microsoft/vcpkg/files/4750305/h5cpp.zip)

**Comment by JackBoosY** [comment]
Please reopen this PR if you continue it.

**Pull request closed without merging**